osu.Game/Beatmaps/Beatmap.cs

```diff
@@ -26,11 +27,26 @@ public double BPMAt(double time)
             return 60000 / BeatLengthAt(time);
         }
 
-        public double BeatLengthAt(double time)
+        public double BPMMultiplierAt(double time)
         {
             ControlPoint overridePoint;
             ControlPoint timingPoint = TimingPointAt(time, out overridePoint);
-            return timingPoint.BeatLength;
+
+            return overridePoint?.VelocityAdjustment ?? 1;
+        }
+
+        public double BeatLengthAt(double time, bool addVelocity = true)

```

Comment:
This is likely to be the default use case, with multilyVelocity = true. I would go the other way around, or perhaps make it local to the use cases where it's == false. Fair point about the naming though.